homeassistant/components/amcrest/config_flow.py

```diff
@@ -0,0 +1,233 @@
+                            CONF_STREAM_SOURCE, DEFAULT_STREAM_SOURCE
+                        ),
+                    ): vol.In(STREAM_SOURCE_LIST),
+                    vol.Optional(
+                        CONF_FFMPEG_ARGUMENTS,
+                        default=self.config_entry.options.get(
+                            CONF_FFMPEG_ARGUMENTS, DEFAULT_FFMPEG_ARGUMENTS
+                        ),
+                    ): str,
+                    vol.Optional(
+                        CONF_RESOLUTION,
+                        default=self.config_entry.options.get(
+                            CONF_RESOLUTION, DEFAULT_RESOLUTION
+                        ),
+                    ): vol.All(vol.In(RESOLUTION_LIST)),
+                    vol.Optional(
+                        CONF_AUTHENTICATION,
+                        default=self.config_entry.options.get(
+                            CONF_AUTHENTICATION, DEFAULT_AUTHENTICATION
+                        ),
+                    ): vol.All(vol.In(AUTHENTICATION_LIST)),
+                    vol.Optional(
+                        CONF_CONTROL_LIGHT,
+                        default=self.config_entry.options.get(
+                            CONF_CONTROL_LIGHT, DEFAULT_CONTROL_LIGHT
+                        ),
+                    ): bool,
+                    vol.Optional(
+                        CONF_SCAN_INTERVAL,
+                        default=self.config_entry.options.get(
+                            CONF_SCAN_INTERVAL, DEFAULT_SCAN_INTERVAL
+                        ),
+                    ): int,
+                    vol.Optional(
+                        CONF_BINARY_SENSORS,
+                        default=self.config_entry.options.get(CONF_BINARY_SENSORS, []),
+                    ): cv.multi_select(
+                        {
+                            CONF_BINARY_SENSOR_AUDIO_DETECTED: CONF_BINARY_SENSOR_AUDIO_DETECTED,
+                            CONF_BINARY_SENSOR_AUDIO_DETECTED_POLLED: CONF_BINARY_SENSOR_AUDIO_DETECTED_POLLED,
+                            CONF_BINARY_SENSOR_MOTION_DETECTED: CONF_BINARY_SENSOR_MOTION_DETECTED,
+                            CONF_BINARY_SENSOR_MOTION_DETECTED_POLLED: CONF_BINARY_SENSOR_MOTION_DETECTED_POLLED,
+                            CONF_BINARY_SENSOR_ONLINE: CONF_BINARY_SENSOR_ONLINE,
+                        }
+                    ),
+                    vol.Optional(
+                        CONF_SENSORS,
+                        default=self.config_entry.options.get(CONF_SENSORS, []),
+                    ): cv.multi_select(
+                        {
+                            CONF_SENSOR_PTZ_PRESET: CONF_SENSOR_PTZ_PRESET,
+                            CONF_SENSOR_SDCARD: CONF_SENSOR_SDCARD,
+                        }
+                    ),
+                    vol.Optional(
+                        CONF_EVENTS,
+                        default=self.config_entry.options.get(CONF_EVENTS, str),
```

Comment:
I agree
in fact only those indicated can be used only as binary sensors, it is not a choice that is left to the user
( from the docs audio_detected, audio_detected_polled, motion_detected, motion_detected_polled, online )

Reply:
I thought that binary_sensors were going to disappear from that integration because a rule of HA integrations is that an integration should not « convert » events to entities and forward the events instead (and the end-user can have a boolean input whose value updates with automations catching the events if she wants).

Also, the configuration under CONF_EVENTS is not the binary_sensor list, AFAICT

Reply:
The breaking change of converting _motion_detected_ from a binary sensor to an event warrants a serious discussion. An entire camera-based security system will fail to alert to intrusions because  the breaking change requiring existing automations to be rewritten was missed.

If this type of change is demanded, then at the very least it needs to be deprecated with accompanying alerts over a period of time.  I should also point out that practically every camera integration uses binary sensors rather than events.   This includes Abode, Android IPcam, August, Axis, Blink, Nest, ONVIF, Ring and Skybell.   The only camera/doorbell integration I found that uses events was the Doorbird.

Going forward, using events and not creating any NEW binary sensors makes perfect sense.   Allowing the existing binary sensors to also be triggered as events in parallel also seems sensible. 

It sounds like a lot of us are using Dahua devices with this Amcrest integration.   It should be the other way around.   How about a new Dahua integration that can be used by all the devices that share its API, without all the baggage and breaking changes to updating this Amcrest code?

Reply:
I agree not to turn binary sensors into events.
It takes careful thought and a PR for just that.
I disagree with creating a clone of an integration, it would just be confusing.